stl/inc/format

```diff
@@ -3397,11 +3397,15 @@ using wformat_args = basic_format_args<wformat_context>;
 
 template <class _Context = format_context, class... _Args>
 _NODISCARD auto make_format_args(_Args&&... _Vals) {
+    static_assert((_Has_formatter<_Args, _Context> && ...),
+        "Cannot format an argument. To make type T formattable, provide a formatter<T> specialization.");

```

Comment:
Would it be helpful to specify the section of the Standard like other static assertions?

Reply:
This diagnostic will be less useful than the current one, because it won't provide what specializations were tried (with MSVC)

Reply:
> This diagnostic will be less useful than the current one, because it won't provide what specializations were tried (with MSVC)

Based on my local testing, it seems that this PR does not prevent MSVC from emitting error C2665 ("no overloaded function could convert all the argument types"), but it makes MSVC emit error C2338 ("static_assert failed") before that. So this PR makes MSVC produce strictly more information.

Reply:
Oddly our quick godbolt testing showed the opposite. We're probably wrong though. I'll test things using the actual code in this PR tomorrow.

Reply:
You are correct, this does not make diagnostics worse